common/flatpak-dir.c

```diff
@@ -5466,6 +5588,33 @@ flatpak_dir_get_system_default (void)
   return flatpak_dir_new (path, FALSE);
 }
 
+GPtrArray *
+flatpak_dir_get_system_list (void)
+{
+  g_autoptr(GPtrArray) result = NULL;
+  GPtrArray *locations = NULL;
+  FlatpakDir *default_dir = NULL;
+  int i;
+
+  locations = flatpak_get_system_base_dir_locations ();
+  if (locations == NULL)
+    return NULL;
```

Comment:
Good point. I'll add a GError to the method and propagate that to wherever is needed